Sources/SCTE35/HelperConverter/BitConverter.swift

```diff
@@ -87,56 +166,168 @@ class BitConverter {
         return stringToReturn
     }
 
+    static func bits(fromUMID umid: String) -> [Bit]? {
+        guard umid.replacingOccurrences(of: ".", with: "").count == 64 else { return nil}
+
+        let hexParts = umid.split(separator: ".")
+        guard hexParts.count == 8 else { return nil }
+
+        var bits = [Bit]()
+        for hexChar in hexParts {
+            guard let integer = UInt32(hexChar, radix: 16) else {
+                return nil
+            }
+            bits.append(contentsOf: BitConverter.bits(from: integer))
+        }
+        return bits
+    }
+
     static func isanString(fromBits bits: [Bit]) -> String? {
-        guard bits.count == 96 else { return nil }
-        let nibbles = stride(from: 0, to: bits.count, by: 4).map { (last) -> [Bit] in
-            return Array(bits[last..<last+4])
+        let containsVersion: Bool
+        if bits.count == 96 {
+            containsVersion = true
+        } else if bits.count == 64 {
+            containsVersion =  false
+        } else {
+            // number of bits does not represent a valid ISAN
+            return nil
         }
 
-        var stringToReturn = ""
-        let indicesForDash = [3, 7, 11, 19]
-        for (index, nibble) in nibbles.enumerated() {
+        var isanIntegers = [Int]()
+        for index in stride(from: bits.startIndex, to: bits.endIndex, by: 4) {
+            let nibble = Array(bits[index..<index+4])
             let intValue = BitConverter.integer(fromBits: nibble)
-            let char = String(format: "%01X", intValue)
-            stringToReturn.append(char)
-            // TODO: - use page 9 of SCTE Documentation [ISO 15706-2] to find check characters
-            if indicesForDash.contains(index) {
-                stringToReturn.append("-")
-            } else if index == 15 {
-                stringToReturn.append("-?-")
-            } else if index == 23 {
-                stringToReturn.append("-?")
-            }
+            isanIntegers.append(intValue)
         }
-        return stringToReturn
+
+        // There must always be a first check character for the root and episode segments
```

Comment:
the first and second check characters are now computed and included in the returned ISAN string.  Also this method can created both the older ISAN and the newer versioned ISAN string.